sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubMessageWithAttributesAndMessageIdAndOrderingKeyCoder.java

```diff
@@ -42,8 +42,9 @@
   // A message's attributes can be null.
   private static final Coder<Map<String, String>> ATTRIBUTES_CODER =
       NullableCoder.of(MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()));
-  // A message's messageId cannot be null
-  private static final Coder<String> MESSAGE_ID_CODER = StringUtf8Coder.of();
+  // A message's messageId can only be null when the message is an outgoing message (i.e. to be
+  // published). Incoming messages will always have a non-null messageId
+  private static final Coder<String> MESSAGE_ID_CODER = NullableCoder.of(StringUtf8Coder.of());
```

Comment:
This is a breaking change as it adds a byte to the encoded message. Why is it needed?

Reply:
This would be needed to allow for use of this coder for outgoing messages, where the messageId field must be null to allow publishing.